### PR TITLE
Simplify TS reload logic

### LINKED ISSUE #3164: Simplify server reloading logic
https://github.com/microsoft/vscode/issues/45774#issuecomment-373423895 <- seems much neater way then what we are doing

### PR DESCRIPTION
**Pull request by matklad**
Fixes #3164

### CODE REVIEW

**Comment by matklad** [comment]
bors r+
🤖

**Comment by matklad** [comment]
npm is not feeling great, merging manually

https://status.npmjs.org/incidents/5qxhhcx0vtlc

File: editors/code/src/inlay_hints.ts [hunk 1]
```diff
@@ -79,6 +92,8 @@ class HintsUpdater {
 
     async refresh() {
         if (!this.enabled) return;
+        console.log("freshin!");
```
Comment (lnicola) [inline editors/code/src/inlay_hints.ts]: Intended?

**Pull request merged**